### ISSUE TITLE
Why do we reimplement models for equistore and not just directly use pytorch or scikit-learn?

### ISSUE THREAD

**Issue by agoscinski** [lab-cosmo/equistore#155]
Need a wiki entry to answer this question extensively.

**Comment by PicoCentauri** [comment]
Thanks for this important issue. We should explain this also in the equistore documentation. Most likely in the explanation section. Maybe I start giving a quick answer here:

The reason is that we try to use the fast implementation of torch and numpy as much as possible and only dispatch our data to the specific ones. Most of our operations are wrappers that check the metadata and then distribute the task either to numpy or to pytorch. This of course creates some convience but overhead which we try to keep as small as possible.

Another reason is that we have gradients and maybe want to fit only in values but predict on energies. Handling those is not directly supported wether by sklearn not torch.
